extensions/quartz/runtime/src/main/java/io/quarkus/quartz/runtime/QuartzScheduler.java

```diff
@@ -302,7 +303,17 @@ public QuartzScheduler(SchedulerContext context, QuartzSupport quartzSupport, Sc
                                         scheduled, oldTrigger.getKey().getName());
                             }
                         }
-                        scheduledTasks.put(identity, new QuartzTrigger(trigger, invoker,
+                        scheduledTasks.put(identity, new QuartzTrigger(trigger.getKey(),
```

Comment:
I'm a bit confused ;-). So why do we need to obtain the trigger dynamically?

Reply:
Because the instance only contains information about the 0th run

Reply:
Hm, my understanding of the quartz API/javadoc was that the trigger instance is updated after every execution...

Reply:
I don't know the Quartz internals, but the linked issue clearly shows we are not doing the proper thing

Reply:
I do not dispute that but I'd like to know what's going on because it could break something later... BTW I don't know quartz internals either ;-)

Reply:
Uf, it seems that for some reason quartz does not work with the original trigger instance but with a clone instead??!!! :fearful: 

@machi1990 it would be good to try to find out whether it's actually a feature or a bug because I don't see it documented anywhere...

Reply:
Quartz always update the trigger information in the corresponding store e.g for db stores, the corresponding trigger table will get updated.